observability-test/spanner.ts

```diff
@@ -522,6 +538,226 @@ describe('ObservabilityOptions injection and propagation', async () => {
     done();
   });
 
+  afterEach(async () => {
+    await injectedTracerProvider.forceFlush();
+    injectedTraceExporter.reset();
+  });
+
+  let database: Database;
+  beforeEach(() => {
+    const instance = spanner.instance('instance');
+    database = instance.database('db');
+  });
+
+  describe('Transaction', () => {
```

Comment:
What extra do we have in 'Transaction' class if we are adding a transaction block here. ?

Reply:
So this is where we test against MockSpanner and if I recall correctly we want to ensure that every class runs through:
* customary code that a customer would write using MockSpanner
* unit tests that exorcise both error and non-error cases
hence this block.

Reply:
I mean we decided to use mockspanner for the observability tests. So why do we need separate transaction test class ? I know we had it earlier, till the time we decided to use mock spanner for these tests.

Reply:
Notice that inside the ones for MockSpanner, we can't easily test out error cases? That's the purpose of the dual tests.